parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -655,7 +660,20 @@ ensureNameLookupForBranchHash getDeclType mayFromBranchHash toBranchHash = do
                   pure $ toNamedRef (name, refWithCT)
               save (termNameAddsWithCT, toNamedRef <$> termNameRemovals) (toNamedRef <$> typeNameAdds, toNamedRef <$> typeNameRemovals)
         )
+      depMounts <- Projects.inferDependencyMounts toBranch <&> fmap (first (coerce @_ @PathSegments . Path.toList))
+      -- Ensure all of our dependencies have name lookups too.
+      for_ depMounts \(_path, depBranchHash) -> do
+        -- TODO: see if we can find a way to infer a good fromHash for dependencies
```

Comment:
It would still be good to do this, but the new shallower name lookups (without transitive deps) are much faster to generate now so it should be okay in the meantime.